### ISSUE TITLE
Get-MailboxPermission

### ISSUE THREAD

**Issue by Bloggzy** [invictus-ir/Microsoft-Extractor-Suite#121]
Hey! I love the project, please keep up the excellent work. This isn't so much an issue as a suggestion, if that's ok?

It relates to: Get-MailboxPermission

During a recent investigation, we found value in enumerating the different mailboxes within the environment and seeing what users had permissions over those mailboxes (as the investigation related to someone accessing mailboxes belonging to others). 

My suggestion would be to add some functionality to do the same thing, if it does not already exist?

Here is the code that I used, I'm not suggesting this is the best way of doing it (at all), but just to illustrate what I was trying to achieve:

```powershell
$mailboxes = Get-Mailbox -ResultSize Unlimited -Filter 'RecipientTypeDetails -eq "UserMailbox"'
$results = @() 
foreach ($mailbox in $mailboxes) {
    # Fetch permissions for the current mailbox
    $permissions = Get-MailboxPermission -Identity $mailbox.alias -ResultSize Unlimited | 
                   Where-Object {($_.IsInherited -eq $false) -and ($_.User -ne "NT AUTHORITY\SELF") -and ($_.AccessRights -contains "FullAccess")} |
                   Select-Object @{Name="Mailbox";Expression={$mailbox.Alias}}, User, AccessRights
    # Add the permissions to the results array
    $results += $permissions
}
# Export the results to a CSV file
$results | Export-Csv -Path ".\MailboxPermissions.csv" -NoTypeInformation -Encoding UTF8

```

**Comment by JoeyInvictus** [comment]
Hi, glad you like the project! We actually already have a function ready for this! We still need to do some additional work and testing, but it will be included in the 3.0.0 version, which we hope to release in the first week of January.

**Comment by Bloggzy** [comment]
Awesome news, nice one, looking forward to it.
